### PR TITLE
Add scripts to allow bridging of Arbitrum fees to mainnet and wd them in L1

### PR DESCRIPTION
**Pull request by petrovska-petro**
Tackles #253 

For the run test refer to this [ticket](https://github.com/Badger-Finance/badger-ape/issues/849). Here brought the test JSON file for the records as well.

From Arb to mainnet commands: 

```
brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/bridge_out_erc20

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/remove_liq_swapr

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/remove_liq_sushiswap

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/wd_one_coin_curve

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/swap_swpr_for_eth

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/vanilla transfer main true {BADGER || WETH}

brownie run --network arbitrum-main badger/bridge_out_arbitrum_fees/bDXS_wd
```

Command in mainnet to wd the bridge over tokens after a week:

```
brownie run --network arbitrum-main bridge_arbitrum/mainnet_args_generation main {introduce_batchNum} {introduce_index}

brownie run bridge_arbitrum/wd_on_mainnet main {msig_withdrawing} {token_to_wd} {introduce_batchNum} {introduce_index}
```

Note that `NO_OUTBOX_ENTRY` revert msg will show-up locally unless that we do not wait the 7d window + some extra time as the outbox entries are created in batches and sharp 7d may not precise

### CODE REVIEW

File: scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from rich.console import Console
+from brownie import interface, accounts, Contract
+from helpers.addresses import registry
+
+CONSOLE = Console()
+
+# tokens to bridge out
+tokens_out = ["CRV", "USDT", "SUSHI", "WBTC"]
```
Comment (jorijnsmit) [inline scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:8]: isnt there also a lot of $weth being withdrawn from the lp positions?

File: scripts/badger/bridge_out_arbitrum_fees/remove_liq_swapr.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from rich.console import Console
+from brownie import interface, chain, Contract, accounts
+from helpers.addresses import registry
+
+CONSOLE = Console()
+
+# dxs tokens to liquidated
+dxs_tokens = ["dxsWbtcWeth", "dxsBadgerWeth"]
```
Comment (jorijnsmit) [inline scripts/badger/bridge_out_arbitrum_fees/remove_liq_swapr.py:8]: i also see dxsSwprWeth (`0xA66b20912cBEa522278f3056B4aE60D0d3EE271b`) and dxsIbbtcWeth (`0x6a060a569e04a41794d6b1308865a13F27D27E53`) balances on the msig.

File: scripts/badger/bridge_out_arbitrum_fees/wd_on_mainnet.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import json
+import os
+from brownie import interface
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import registry
+
+
+def main(msig="techops_multisig", token_bridge="CRV", batchNum=0, index=0):
+    safe = GreatApeSafe(registry.eth.badger_wallets[msig])
```
Comment (jorijnsmit) [inline scripts/badger/bridge_out_arbitrum_fees/wd_on_mainnet.py:9]: ah do i understand correctly techops is withdrawing for treasury ops?

**Review by jorijnsmit — CHANGES_REQUESTED** [review]
obviously i cant run this stuff on fork but did a code review and left comments.

also, i see some `bdxsSwaprWeth` made its way to the msig, shouldnt that also be withdrawn and unwrapped?

File: scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py (hunk 1 above)
Comment (jorijnsmit) [reply scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:8]: and some $swapr?

File: scripts/badger/bridge_out_arbitrum_fees/json_withdrawal_payload/batchNum_6505_index_0.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "calldataForL1": "2e567b36000000000000000000000000d533a949740bb3306d119cc777fa900ba034cd52000000000000000000000000f6bc36280f32398a031a7294e81131aee787d17800000000000000000000000086cbd0ce0c087b482782c181da8d191de18c82750000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000dc500000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000000",
+    "l1Block": 14044252,
+    "l1Dest": "0xa3A7B6F88361F48403514059F1F16C8E78d60EeC",
+    "l2Block": 4831404,
+    "l2Sender": "0x09e9222E96E7B4AE2a407B98d48e330053351EEe",
+    "path": 1,
+    "proof": [
+        "1ee4300ff66ded3b8c76ac7e8388f64ab40cf8d8ef303099fa5f9899828446af"
+    ],
+    "timestamp": 1642705333
```
[the remark this answers is not in the document]
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/json_withdrawal_payload/batchNum_6505_index_0.json:12]: yep, just for the records adding it in here

File: scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        if token_balance > 0:
+            calldata_approve = token.approve.encode_input(
+                gateway.address, token_balance
+            )
+
+            calldata_bridging = gateway.outboundTransfer.encode_input(
+                registry.arbitrum.treasury_tokens[f"{key}"],
+                registry.eth.badger_wallets.treasury_ops_multisig,
+                token_balance,
+                bytes("", encoding="UTF-8"),
```
[the remark this answers is not in the document]
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:38]: on the test we used `bytes("", encoding="UTF-8")` last time, but both b'' and this, will generate the empty byte that we need

File: scripts/badger/bridge_out_arbitrum_fees/wd_on_mainnet.py (hunk 3 above)
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/wd_on_mainnet.py:9]: just a msig as default from last test, but yeah makes more sense trops 🤝

File: scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py (hunk 1 above)
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:8]: weth will stay in Arb for gas ops, i think we can send to techops and unwrap there

will add some method to swap `SWPR` for another asset

File: scripts/badger/bridge_out_arbitrum_fees/remove_liq_swapr.py (hunk 2 above)
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/remove_liq_swapr.py:8]: good catch!

File: scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py (hunk 1 above)
Comment (petrovska-petro) [reply scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:8]: thinking that for ops cost the value wd in `weth` is going to be quite a lot considering the tx fee in Arb

perhaps worth sending some to techops and rest to mainnet?

Comment (jorijnsmit) [reply scripts/badger/bridge_out_arbitrum_fees/bridge_out_erc20.py:8]: yea techops makes sense for gas

**Comment by jorijnsmit** [comment]
too bad i cant run this on fork! guess that will all be solved after this issue though :)

code looks good, all my comments were processed. only thing is i would change the order here a bit; think swapping the first and last action should do the trick:
Screenshot 2022-04-08 at 16 23 47

**Review by jorijnsmit — APPROVED** [review]

**Review by jalbrekt85 — APPROVED** [review]
cannot run locally so manually reviewing
-all addresses used look good 
-remove liq/swap/wd functions look to be called correctly

**Pull request merged**